tests/tests_lite/plugins/precision/test_double_integration.py

```diff
@@ -0,0 +1,55 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Integration tests for double-precision training."""
+
+import torch
+import torch.nn as nn
+from tests_lite.helpers.models import BoringLite
+
+
+class BoringDoubleModule(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.layer = torch.nn.Linear(32, 2)
+        self.register_buffer("complex_buffer", torch.complex(torch.rand(10), torch.rand(10)), False)
+
+    def forward(self, x):
+        assert x.dtype == torch.float64
+        assert torch.tensor([0.0]).dtype == torch.float64
+        assert torch.tensor([0.0], dtype=torch.float16).dtype == torch.float16
```

Comment:
why this check?  If this doesn't hold this is an issue with torch not on our side.